sequence_task_1/student/include/impl.h

```diff
@@ -15,36 +16,101 @@ bool operator==(const Song& lhs, const Song& rhs)
     return lhs.name == rhs.name;
 }
 
-template<class Container>
-class StaticPlaylist
-{
+template <class Container>
+class StaticPlaylist {
 public:
     /** @todo Member traits */
+    using value_type = typename Container::value_type;
+    using difference_type = typename Container::difference_type;
+    using size_type = typename Container::size_type;
+    using reference = typename Container::reference;
+    using const_reference = typename Container::const_reference;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using reverse_iterator = typename Container::reverse_iterator;
 
     /** @todo Iterators */
+    const_iterator begin() const
+    {
+        return m_tracklist.begin();
+    }
+
+    const_iterator end() const
+    {
+        return m_tracklist.end();
+    }
+
+    const_iterator cbegin() const
+    {
+        return m_tracklist.cbegin();
+    }
+
+    const_iterator cend() const
+    {
+        return m_tracklist.cend();
+    }
 
     /** @todo Constructor from any reversible sequence container */
+    template <typename OtherContainer>
+    StaticPlaylist(OtherContainer c)
```

Comment:
It is better to separate copy and move.
In case of lvalue, you will perform redundant copy here.

Reply:
Done